templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -364,76 +349,97 @@ contents:
         # no config was changed, no need to reload
         return
       fi
-      nm_config_changed=0
-      
-      echo "Reloading NetworkManager after configuration changes..."
 
-      # recycle network, so that existing profiles and priorities are re-evaluated
-      nmcli network off
-
-      # wait for no devices to show as connected
-      echo "Waiting for devices to disconnect..."
-      if ! timeout 60 bash -c "while nmcli -g DEVICE,STATE d | grep -v :unmanaged; do sleep 5; done"; then
-        echo "Warning: NetworkManager did not disconnect all devices"
-      fi
-      
-      # reload profiles and set networking back on
+      # reload profiles
       nmcli connection reload
-      nmcli network on
-      
-      # restart NetworkManager so that we can wait on `nm-online -s`
-      systemctl restart NetworkManager
 
-      # Wait until all profiles auto-connect
-      echo "Waiting for profiles to activate..."
-      if nm-online -s -t 60; then
-        echo "NetworkManager has activated all suitable profiles after reload"
-      else
-        echo "Warning: NetworkManager has not activated all suitable profiles after reload"
-      fi
+      # precautionary sleep of 10s (default timeout of NM to bring down devices)
+      sleep 10
+    
+      # After reload, devices that were already connected should connect again
+      # if any profile is available. If no profile is available, a device can
+      # remain disconnected and we have to explicitly connect it so that a
+      # profile is generated. This can happen for physical devices but should
+      # not happen for software devices as those always require a profile.
+      for dev in $@; do
+        # Only attempt to connect a disconnected device
+        local connected_state=$(nmcli -g GENERAL.STATE device show "$dev" || echo "")
+        if [[ "$connected_state" =~ "disconnected" ]]; then
+          # keep track if a profile by the same name as the device existed 
+          # before we attempt activation
+          local named_profile_existed=$([ -f "${NM_CONN_PATH}/${dev}" ] || [ -f "${NM_CONN_PATH}/${dev}.nmconnection" ] && echo "yes")
+          
+          for i in {1..10}; do
+              echo "Attempt $i to connect device $dev"
+              nmcli device connect "$dev" && break
+              sleep 5
+          done
 
-      # Check if we have any type of connectivity
-      if nm-online -t 0; then
-        echo "NetworkManager has connectivity after reload"
-      else
-        echo "Warning: NetworkManager does not have connectivity after reload"
-      fi
+          # if a profile did not exist before but does now, it was generated
+          # but we want it to be ephemeral, so move it back to /run
+          if [ ! "$named_profile_existed" = "yes" ]; then
+            local dst="/run/NetworkManager/system-connections/"
+            MANAGED_NM_CONN_FILES=("${NM_CONN_PATH}/${dev}" "${NM_CONN_PATH}/${dev}.nmconnection")
```

Comment:
Does using this global var here risk to interfere in other locations with copy_nm_conn_files / rm_nm_conn_files, as in might this introduce order of operation issues due to a shared global variable?

So far, I think that we run update_nm_conn_files before any of the rm_ or copy_ operations, so we're good?

Reply:
Yeah, we always have to consider the value `MANAGED_NM_CONN_FILES` and update it manuall or via `update_nm_conn_files` before calling `rm_` or `copy_`